src/main/java/com/streamr/client/dataunion/DataUnion.java

```diff
@@ -0,0 +1,166 @@
+        } catch (Exception e) {
+            log.error(e.getMessage());
+            throw new RuntimeException(e);
+        }
+    }
+
+    private DataUnionMainnet mainnet;
+    private DataUnionSidechain sidechain;
+
+    //use DataUnionClient to instantiate
+    protected DataUnion(DataUnionMainnet mainnet, DataUnionSidechain sidechain) {
+        this.mainnet = mainnet;
+        this.sidechain = sidechain;
+    }
+
+    public boolean waitForDeployment(long sleeptime, long timeout) throws Exception {
+        String code = waitForCodeAtAddress(sidechain.getContractAddress(), getWeb3j(sidechain), sleeptime, timeout);
+        return code != null && !code.equals("0x");
+    }
+
+    public String mainnetAddress(){
+        return mainnet.getContractAddress();
+    }
+
+    public String sidechainAddress(){
+        return sidechain.getContractAddress();
+    }
+
+    public boolean isDeployed() throws Exception {
+        //this will check the condition only once:
+        return waitForDeployment(0,0);
+    }
+
+    public void sendTokensToBridge() throws Exception {
+        mainnet.sendTokensToBridge().send();
+    }
+
+    public BigInteger waitForEarningsChange(final BigInteger initialBalance, long sleeptime, long timeout) throws Exception {
+        Web3jUtils.Condition earningsChange = new Web3jUtils.Condition(){
+            @Override
+            public Object check() throws Exception {
+                BigInteger bal = sidechain.totalEarnings().send().getValue();
+                if(!bal.equals(initialBalance))
+                    return bal;
+                return null;
+            }
+        };
+        return (BigInteger) waitForCondition(earningsChange, sleeptime, timeout);
+   }
+
+    public EthereumTransactionReceipt joinMembers(String ... members) throws Exception {
+        ArrayList<Address> memberAddresses = new ArrayList<Address>(members.length);
+        for (String member : members) {
+            memberAddresses.add(new Address(member));
+        }
+        return new EthereumTransactionReceipt(sidechain.addMembers(new DynamicArray<Address>(Address.class, memberAddresses)).send());
+    }
+
+    //amount == 0 means withdrawAll!
+    public EthereumTransactionReceipt withdraw(BigInteger withdrawerPrivateKey, String to, BigInteger amount) throws Exception {
+        return withdraw(Credentials.create(ECKeyPair.create(withdrawerPrivateKey)), to, amount);
+    }
+
+    public EthereumTransactionReceipt withdraw(String withdrawerPrivateKey, String to, BigInteger amount) throws Exception {
+        return withdraw(Credentials.create(withdrawerPrivateKey), to, amount);
+    }
+
+    protected EthereumTransactionReceipt withdraw(Credentials from, String to, BigInteger amount) throws Exception {
+        byte[] req = createWithdrawRequest(from.getAddress(), to, amount);
+        byte[] sig = signWithdraw(from, req);
+        if(amount.equals(BigInteger.ZERO))
```

Comment:
Use curly braces in if/for/etc constructs. Without it is a potential source for bugs in the future.